core/base/matrix_data.hpp

```diff
@@ -238,6 +240,24 @@ struct matrix_data {
         this->ensure_row_major_order();
     }
 
+    /**
+     * Initializes a matrix from a range.
+     *
+     * @return data  range used to initialize the matrix
+     */
+    template <typename Accessor>
+    matrix_data(const range<Accessor> &data)
```

Comment:
The doxygen looks incomplete/incorrect here. it should be `@param` and a `@tparam` could be added.